Libraries/Components/Touchable/Touchable.js

```diff
@@ -325,9 +325,17 @@ const TouchableMixin = {
       evt.dispatchConfig = {};
       if (myTag === evt.tag) {
         if (evt.eventType === 'focus') {
-          cmp.touchableHandleActivePressIn && cmp.touchableHandleActivePressIn(evt);
+          if (cmp.touchableHandleActivePressIn) {
```

Comment:
I had the impression that developers using RN would not use `Touchable` directly, but use `TouchableOpacity`, `TouchableHighlight`, etc., which implement `onTouchableHandleActivePressIn` (and `Out`).  So it seems to me that those files also need to change, yes?  And the way this has been done, it appears that components implementing `Touchable` methods aren't allowed to handle both press and focus events.  It seems to me that that would be a big breaking change for Apple TV, at the very least.

Reply:
I'll double check the code soon. If I'm remembering correctly, this base mixin is included into TouchableOpacity, etc and is used as the single event handler entry point. This code then defers to `touchableHandleActivePressIn` and `touchableHandleFocus`. `touchableHandleActivePressIn` has default implementations in some of those components, e.g. TouchableBounce provides a `touchableHandleActivePressIn` which does the bounce effect and then calls `props.onPressIn` if it was provided. For focus, the idea in this PR would be that Touchable itself has the default implementation of `touchableHandleFocus` which then defers to `onFocus` if it was indeed provided. This was done mostly because some custom Touchables might want to have a style/effect change on focus, but I didn't want to make that opinion just yet, though I can definitely see some sort of default style being desirable--wanted to keep the scope narrow to make the discussion easier.

I'll circle back after I've had a chance to review the codebase again, but do let me know if I'm off base. 👍

Reply:
Also to clarify, this PR as it stands does not remove the behavior of onPressIn being called for focus events, it just deprecates it so that it could be removed in a future release so that a true distinction could be made between onPressIn and onFocus, and in follow up PRs potentially opening up onFocus to other platforms than just TV

Reply:
@dlowder-salesforce I checked the code again, my above description appears correct. Lmk I'm mistaken.

Reply:
Thanks @jayphelps .... sorry for my slowness.  I read it over again, and had a good discussion about it with @matthargett .  Looks good.